File: packages/shared/src/types/fax.ts
import type { UUID, FaxDirection, FaxStatus } from './common';

export interface Fax {
  id: UUID;
  direction: FaxDirection;
  status: FaxStatus;
  from_number: string;
  to_number: string;
  pages: number;
  file_url: string | null;
  patient_id: UUID | null;
  patient_name: string | null;
  matched_by: string | null;
  ocr_text: string | null;
  created_at: string;
}

export interface FaxSendRequest {
  to_number: string;
  patient_id?: UUID;
  file_url?: string;
}

File: apps/api/app/schemas/fax.py
from pydantic import BaseModel, Field


class FaxSendRequest(BaseModel):
    to_number: str = Field(min_length=10)
    patient_id: str | None = None
    file_url: str | None = None
    pages: int = Field(default=1, ge=1)
    ocr_text: str | None = None


class FaxMatchRequest(BaseModel):
    patient_id: str


class FaxOut(BaseModel):
    id: str
    direction: str
    status: str
    from_number: str
    to_number: str
    pages: int
    file_url: str | None
    patient_id: str | None
    patient_name: str | None = None
    matched_by: str | None
    ocr_text: str | None
    created_at: str

    model_config = {"from_attributes": True}


class FaxListOut(BaseModel):
    data: list[FaxOut]
    total: int
    page: int
    page_size: int
